meta-resin-common/classes/host-update-resin.bbclass

```diff
@@ -16,14 +16,22 @@ RESIN_HUP_BUNDLE ?= "${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.rootfs.resinhup-tar"
 
 QUIRK_FILES ?= "etc/hostname etc/hosts etc/resolv.conf"
 
+# check if we are running on a poky version which deploys to IMGDEPLOYDIR instead of DEPLOY_DIR_IMAGE (poky morty introduced this change)
```

Comment:
I would recommend having this in the CMD block itself. This is because yocto parses the recipe multiple times so this block will be triggered multiple times for no reason. Having it in the CMD function will only make it run when it is needed.